src/system_info/system_info_network_ivi.cc

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2014 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "system_info/system_info_network.h"
+
+SysInfoNetwork::SysInfoNetwork()
+    : type_(SYSTEM_INFO_NETWORK_UNKNOWN) {
+  PlatformInitialize();
+}
+
+void SysInfoNetwork::PlatformInitialize() {
+  conn_ = system_info::GetDbusConnection();
+  if (!conn_)
+    return;
+  modem_path_ = system_info::OfonoGetModemPath(conn_);
+}
+
+SysInfoNetwork::~SysInfoNetwork() {
+  if (!conn_)
+    return;
+  g_dbus_connection_signal_unsubscribe(conn_,
+      cellular_network_type_changed_watch_);
+  g_dbus_connection_close_sync(conn_, NULL, NULL);
+  conn_ = NULL;
+}
+
+void SysInfoNetwork::StartListening() {
+  vconf_notify_key_changed(VCONFKEY_NETWORK_STATUS,
+      (vconf_callback_fn)OnNetworkTypeChanged, this);
+
+  if (!conn_ || !modem_path_)
+    return;
+  cellular_network_type_changed_watch_ = g_dbus_connection_signal_subscribe(
+      conn_, system_info::kOfonoService,
+      system_info::kOfonoNetworkRegistrationIface, "PropertyChanged",
+      modem_path_, NULL, G_DBUS_SIGNAL_FLAGS_NONE,
+      OnCellularNetworkTypeChanged, this, NULL);
+}
+
+void SysInfoNetwork::StopListening() {
+  vconf_ignore_key_changed(VCONFKEY_NETWORK_STATUS,
+      (vconf_callback_fn)OnNetworkTypeChanged);
+
+  if (!conn_)
+    return;
+  g_dbus_connection_signal_unsubscribe(conn_,
+      cellular_network_type_changed_watch_);
+}
+
+void SysInfoNetwork::GetCellularNetworkType() {
+  if (!conn_ || !modem_path_) {
+    type_ = SYSTEM_INFO_NETWORK_UNKNOWN;
+    return;
+  }
+
+  GError* error = NULL;
+  GVariant* var = g_dbus_connection_call_sync(conn_, system_info::kOfonoService,
+      modem_path_, system_info::kOfonoNetworkRegistrationIface,
+      "GetProperties", NULL, NULL, G_DBUS_CALL_FLAGS_NONE, -1, NULL, &error);
+
+  if (!var) {
+    std::cout << "Ofono network GetProperties failed: "
+              << error->message << std::endl;
+    g_error_free(error);
+    type_ = SYSTEM_INFO_NETWORK_UNKNOWN;
+    return;
+  }
+
+  gchar* key;
+  GVariantIter* iter;
+  GVariant* var_val;
+  g_variant_get(var, "(a{sv})", &iter);
+  while (g_variant_iter_next(iter, "{sv}", &key, &var_val))
+    UpdateCellularNetwork(key, var_val);
+
+  g_variant_unref(var);
+  g_free(key);
+  g_variant_iter_free(iter);
+  g_variant_unref(var_val);
+}
+
+void SysInfoNetwork::UpdateCellularNetwork(const gchar* key,
+                                           GVariant* var_val) {
+  if (g_strcmp0(key, "Technology") != 0)
+    return;
+
+  const char* val = g_variant_get_string(var_val, NULL);
+  if (!val) {
+    type_ = SYSTEM_INFO_NETWORK_NONE;
+    return;
+  } else if (g_strcmp0(val, "gsm") == 0) {
+    type_ = SYSTEM_INFO_NETWORK_2G;
```

Comment:
The ofono strings (gsm, edge, etc) are much more useful than translations like 2G, 3G, etc. Not your fault, the Tizen spec needs fixed.